BSP/STM32F413/Src/BSP_CAN.c

```diff
@@ -149,8 +149,11 @@ ErrorStatus BSP_CAN_Write(uint32_t id, uint8_t data[8], uint8_t length) {
     
     gTxMessage.StdId = id;
     gTxMessage.DLC = length;
-	for(int i = 0; i < length; i++){
-        gTxMessage.Data[i] = data[i];
+
+    if(length<=8){
+        for(int i=0; i<length; i++){
+            gTxMessage.Data[i] = data[i];
+        }
```

Comment:
The general practice for PR's in GitHub is that one PR changes one issue in the code. I am guessing you did this to fix the gdb errors in the code which is an issue separate from the RTOS Create Task. It is fine for now, but for later on I just wanted you to know.